### PR TITLE
perf: Attributes Query - loadTree only once

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Build tree just once as it is costly

All tests passing, including the performance test.

### CODE REVIEW

File: packages/app-store/routing-forms/trpc/utils.ts [hunk 1]
```diff
@@ -215,30 +265,15 @@ export async function findTeamMembersMatchingAttributeLogicOfRoute(
...
-          moduleLogger.debug(`Team member ${member.userId} matches attributes logic`);
-          teamMembersMatchingAttributeLogicMap.set(member.userId, result);
-        } else {
-          moduleLogger.debug(`Team member ${member.userId} does not match attributes logic`);
+
+        const result = !!jsonLogic.apply(logic as any, attributesData)
+          ? RaqbLogicResult.MATCH
+          : RaqbLogicResult.NO_MATCH;
+
+        if (result !== RaqbLogicResult.MATCH) {
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/trpc/utils.ts:273]: Switched to early return

File: packages/app-store/routing-forms/trpc/utils.ts [hunk 2]
```diff
@@ -215,30 +265,15 @@ export async function findTeamMembersMatchingAttributeLogicOfRoute(
...
-          }
-        );
-
-        if (result === RaqbLogicResult.MATCH || result === RaqbLogicResult.LOGIC_NOT_FOUND_SO_MATCHED) {
-          moduleLogger.debug(`Team member ${member.userId} matches attributes logic`);
-          teamMembersMatchingAttributeLogicMap.set(member.userId, result);
-        } else {
-          moduleLogger.debug(`Team member ${member.userId} does not match attributes logic`);
+
+        const result = !!jsonLogic.apply(logic as any, attributesData)
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/trpc/utils.ts:269]: Using plain jsonLogic in loop now. Thanks @emrysal 🙏

File: packages/app-store/routing-forms/trpc/utils.ts [hunk 3]
```diff
@@ -215,30 +265,15 @@ export async function findTeamMembersMatchingAttributeLogicOfRoute(
...
-            data: attributesData,
-            beStrictWithEmptyLogic: true,
-          },
-          {
-            // This logic runs too many times as it is per team member and we don't want to spam the console with logs. It might also take a performance hit otherwise
-            logLevel: 2,
-          }
-        );
-
-        if (result === RaqbLogicResult.MATCH || result === RaqbLogicResult.LOGIC_NOT_FOUND_SO_MATCHED) {
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/trpc/utils.ts:235]: Not needed for now LOGIC_NOT_FOUND_SO_MATCHED

**Review by emrysal — APPROVED** [review]
Looks good 👍

**Pull request merged**